### PR TITLE
[Feature] [connector] Remove `Hdfs Connector` and make it compatible with `File Connector`

### LINKED ISSUE #1255: [Feature] [connector] Remove Hdfs Connector and make it compatible with File Connector
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support spark hdfs source

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by simon824**
## Purpose of this pull request
closed  #1255 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by CalvinKirs** [comment]
@wuchunfu hi,PTAL,thx

File: docs/en/spark/configuration/source-plugins/Hdfs.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+| serializer | string | no | json |
+| common-options | string | no | - |
+
+#### options [object]
+
+Custom parameters
+
+#### partition_by [array]
+
+Partition data based on selected fields
```
Comment (wuchunfu) [inline docs/en/spark/configuration/source-plugins/Hdfs.md]: source plugin doesn't seem to need this parameter

File: docs/en/spark/configuration/source-plugins/Hdfs.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+#### path [string]
+
+Output file path, starting with **hdfs://**
+
+#### save_mode [string]
+
+Storage mode, currently supports overwrite, append, ignore and error. For the specific meaning of each mode, see [save-modes](http://spark.apache.org/docs/2.2.0/sql-programming-guide.html#save-modes)
```
Comment (wuchunfu) [inline docs/en/spark/configuration/source-plugins/Hdfs.md]: source plugin doesn't seem to need this parameter

File: seatunnel-connectors/seatunnel-connector-spark-file/src/main/scala/org/apache/seatunnel/spark/source/FileSourceBase.scala [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        case _ => checkResult
+      }
+    } else {
+      checkResult
+    }
+  }
+
+  protected def fileReader(spark: SparkSession, path: String): Dataset[Row] = {
+    val format = config.getString("format")
+    var reader = spark.read.format(format)
```
Comment (wuchunfu) [inline seatunnel-connectors/seatunnel-connector-spark-file/src/main/scala/org/apache/seatunnel/spark/source/FileSourceBase.scala]: The `format` parameter doesn't seem to be described in the documentation

File: seatunnel-connectors/seatunnel-connector-spark-file/src/main/scala/org/apache/seatunnel/spark/source/Hdfs.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import org.apache.seatunnel.common.config.CheckResult
+import org.apache.seatunnel.spark.SparkEnvironment
+import org.apache.spark.sql.{Dataset, Row}
+
+class Hdfs extends FileSourceBase {
+
+  override def checkConfig(): CheckResult = {
+    checkConfigImpl("hdfs://")
+  }
```
Comment (wuchunfu) [inline seatunnel-connectors/seatunnel-connector-spark-file/src/main/scala/org/apache/seatunnel/spark/source/Hdfs.scala]: Wouldn't it be better to extract `hdfs://` as a constant?

File: docs/en/spark/configuration/source-plugins/Hdfs.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+#### serializer [string]
+
+Serialization method, currently supports csv, json, parquet, orc and text
+
+### common options [string]
+
+Source Plugin common parameters, refer to [Source Plugin](./source-plugin.md) for details
```
Comment (wuchunfu) [inline docs/en/spark/configuration/source-plugins/Hdfs.md]: The `result_table_name` parameter seems to be `required` , please check

**Comment by simon824** [comment]
Updated, thanks a lot for your review  @wuchunfu , please check it again when free thanks.

**Review by wuchunfu — DISMISSED** [review]
LGTM

**Comment by CalvinKirs** [comment]
hi, thanks for your contribution, but we already have a `files` source, can you describe the difference between the two

**Comment by simon824** [comment]
> hi, thanks for your contribution, but we already have a `files` source, can you describe the difference between the two

`File source connector` now only supports the prefix `file://`, this pr refers to `File sink` to extend support for hdfs.

**Before**:
image  

**After**:
image

**Comment by CalvinKirs** [comment]
> > hi, thanks for your contribution, but we already have a `files` source, can you describe the difference between the two
> 
> `File source connector` now only supports the prefix `file://`, this pr refers to `File sink` to extend support for hdfs.
> 
> **Before**: image
> 
> **After**: image

If it's just this change, we might be better off making changes to the `files` plugin instead of adding a plugin.

**Comment by simon824** [comment]
>if it's just this change, we might be better off making changes to the files plugin instead of adding a plugin.

@CalvinKirs  make sense , i updated supporting for `hdfs` base on `File sink`, PTAL when free thanks.

**Comment by CalvinKirs** [comment]
@wuchunfu Hi, PTAL again, thx

**Comment by wuchunfu** [comment]
> > > hi, thanks for your contribution, but we already have a `files` source, can you describe the difference between the two
> > 
> > 
> > `File source connector` now only supports the prefix `file://`, this pr refers to `File sink` to extend support for hdfs.
> > **Before**: image
> > **After**: image
> 
> If it's just this change, we might be better off making changes to the `files` plugin instead of adding a plugin.

@simon824 Can the `sink plugin` also be like the `source plugin` , without distinguishing in the code whether it is `file://` or `hdfs://`.

**Comment by simon824** [comment]
@wuchunfu make sense,  i will check it later and achieve it next pr.

**Comment by CalvinKirs** [comment]
> @wuchunfu make sense, i will check it later and achieve it next pr.

It would be even better if it could be implemented in this PR.

**Comment by simon824** [comment]
> > @wuchunfu make sense, i will check it later and achieve it next pr.
> 
> It would be even better if it could be implemented in this PR.

All right.

**Comment by simon824** [comment]
Updated, PTAL again when free thanks. @wuchunfu @CalvinKirs

**Comment by wuchunfu** [comment]
@simon824 It looks good on the whole, but the `sink plugin` doesn't seem to need the `common-options` option, please check it. Another point is to test the `source plugin` and `sink plugin` , thank you

**Comment by simon824** [comment]
hi @wuchunfu ,i found  the option`source_table_name` is in `common-options` , do we need it? 
FYI: https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/spark/configuration/sink-plugins/sink-plugin.md

**Comment by wuchunfu** [comment]
> hi @wuchunfu ,i found the option`source_table_name` is in `common-options` , do we need it? FYI: https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/spark/configuration/sink-plugins/sink-plugin.md

@simon824 Oh, maybe I'm mistaken, this seems to be optional.

**Comment by wuchunfu** [comment]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**